app/src/main/kotlin/com/simplemobiletools/dialer/dialogs/SelectContactDialog.kt

```diff
@@ -23,28 +31,115 @@ class SelectContactDialog(val activity: SimpleActivity, contacts: MutableList<Co
+            select_contact_list.adapter = ContactsAdapter(activity, contacts.toMutableList(), select_contact_list, allowLongClick = false) {
                 callback(it as Contact)
                 dialog?.dismiss()
             }
         }
 
         activity.getAlertDialogBuilder()
             .setNegativeButton(R.string.cancel, null)
+            .setOnKeyListener { _, i, keyEvent ->
+                if (keyEvent.action == KeyEvent.ACTION_UP && i == KeyEvent.KEYCODE_BACK) {
+                    backPressed()
+                }
+                true
+            }
             .apply {
-                activity.setupDialogStuff(view, this) { alertDialog ->
+                activity.setupDialogStuff(view, this, R.string.choose_contact) { alertDialog ->
                     dialog = alertDialog
                 }
             }
     }
+
+    private fun setupLetterFastScroller(contacts: List<Contact>) {
+        view.letter_fastscroller.setupWithRecyclerView(view.select_contact_list, { position ->
+            try {
+                val name = contacts[position].getNameToDisplay()
+                val character = if (name.isNotEmpty()) name.substring(0, 1) else ""
+                FastScrollItemIndicator.Text(character.uppercase(Locale.getDefault()))
+            } catch (e: Exception) {
+                FastScrollItemIndicator.Text("")
+            }
+        })
+    }
+
+    private fun configureSearchView() = with(searchView) {
+        updateHintText(context.getString(R.string.search_contacts))
+        searchEditText.imeOptions = EditorInfo.IME_ACTION_DONE
+
+        toggleHideOnScroll(true)
+        setupMenu()
+        setSearchViewListeners()
+        updateSearchViewUi()
+    }
+
+    private fun MySearchMenu.updateSearchViewUi() {
+        getToolbar().beInvisible()
+        updateColors()
+        setBackgroundColor(Color.TRANSPARENT)
+        searchViewAppBarLayout.setBackgroundColor(Color.TRANSPARENT)
+    }
+
+    private fun MySearchMenu.setSearchViewListeners() {
+        onSearchOpenListener = {
+            updateSearchViewLeftIcon(R.drawable.ic_cross_vector)
+        }
+        onSearchClosedListener = {
+            searchEditText.clearFocus()
+            activity.hideKeyboard(searchEditText)
+            updateSearchViewLeftIcon(R.drawable.ic_search_vector)
+        }
+
+        onSearchTextChangedListener = { text ->
+            filterContactListBySearchQuery(text)
+        }
+    }
+
+    private fun updateSearchViewLeftIcon(iconResId: Int) = with(view.findViewById<ImageView>(R.id.top_toolbar_search_icon)) {
+        post {
+            setImageResource(iconResId)
+        }
+    }
+
+    private fun filterContactListBySearchQuery(query: String) {
+        val adapter = view.select_contact_list.adapter as? ContactsAdapter
+        var contactsToShow = contacts
+        if (query.isNotEmpty()) {
+            contactsToShow = contacts.filter { it.name.contains(query, true) }
```

Comment:
Do we need to add some sorting here? 
Just filtering by search string works well. Same in File Manager

Reply:
use the already existing sorting, just filter out some items